components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/webextension/GeckoWebExtension.kt

```diff
@@ -244,15 +246,22 @@ class GeckoWebExtension(
     /**
      * See [WebExtension.registerTabHandler].
      */
-    override fun registerTabHandler(tabHandler: TabHandler) {
+    override fun registerTabHandler(tabHandler: TabHandler, defaultSettings: Settings?) {
 
         val tabDelegate = object : GeckoNativeWebExtension.TabDelegate {
 
+            val trackingPolicy = defaultSettings?.trackingProtectionPolicy
+                ?: EngineSession.TrackingProtectionPolicy.strict()
```

Comment:
See my comment above. Also here we have this outside `onNewTab` whereas in beta we have it inside, but ideally we shouldn't need this. The `GeckoWebExtension` should not need to know about tracking protection policies and just pass settings along.